certmanager/generate_cert.go

```diff
@@ -83,6 +83,32 @@ const (
 // See http://www.alvestrand.no/objectid/2.5.29.17.html.
 var oidSubjectAltName = asn1.ObjectIdentifier{2, 5, 29, 17}
 
+// GenCSR generates a X.509 certificate sign request with the given options.
+func GenCsr(options CertOptions) ([]byte, []byte) {
```

Comment:
As discussed, lets not replicate code. Some refactoring will help keep common code in a function.